src/main/java/net/sacredlabyrinth/phaed/simpleclans/managers/ChatManager.java

```diff
@@ -0,0 +1,150 @@
+                        !allymember.isMutedAlly()).collect(Collectors.toList()));
+                receivers.addAll(clan.getOnlineMembers().stream().filter(allymember ->
+                        !allymember.isMutedAlly()).collect(Collectors.toList()));
+                break;
+            case CLAN:
+                if (!plugin.getSettingsManager().is(CLANCHAT_ENABLE)) {
+                    return;
+                }
+
+                receivers.addAll(clan.getOnlineMembers().stream().
+                        filter(member -> !member.isMuted()).
+                        collect(Collectors.toList()));
+        }
+
+        SCMessage scMessage = new SCMessage(source, channel, clanPlayer, message, receivers);
+
+        for (ChatHandler ch : handlers) {
+            if (ch.canHandle(source)) {
+                ch.sendMessage(scMessage);
+            }
+        }
+    }
+
+    public String parseChatFormat(String format, SCMessage message) {
+        return parseChatFormat(format, message, new HashMap<>());
+    }
+
+    public String parseChatFormat(String format, SCMessage message, Map<String, String> placeholders) {
+        SettingsManager sm = plugin.getSettingsManager();
+        ClanPlayer sender = message.getSender();
+        Channel channel = message.getChannel();
+
+        String leaderColor = sm.getColored(ConfigField.valueOf(message.getChannel() + "CHAT_LEADER_COLOR"));
+        String memberColor = sm.getColored(ConfigField.valueOf(message.getChannel() + "CHAT_MEMBER_COLOR"));
+        String trustedColor = sm.getColored(ConfigField.valueOf(message.getChannel() + "CHAT_TRUSTED_COLOR"));
+
+        String rank = sender.getRankId().isEmpty() ? null : ChatUtils.parseColors(sender.getRankDisplayName());
+        ConfigField configField = ConfigField.valueOf(String.format("%sCHAT_RANK",
+                message.getSource() == SPIGOT ? message.getChannel() : message.getSource()));
+        String rankFormat = (rank != null) ? sm.getColored(configField).replace("%rank%", rank) : "";
+
+        if (placeholders != null) {
+            for (Map.Entry<String, String> e : placeholders.entrySet()) {
+                format = format.replace("%" + e.getKey() + "%", e.getValue());
+            }
+        }
+
+        String parsedFormat = ChatUtils.parseColors(format)
+                .replace("%clan%", Objects.requireNonNull(sender.getClan()).getColorTag())
+                .replace("%nick-color%",
+                        (sender.isLeader() ? leaderColor : sender.isTrusted() ? trustedColor : memberColor))
+                .replace("%player%", sender.getName())
+                .replace("%rank%", rankFormat)
+                .replace("%message%", message.getContent());
+
+        return getPluginManager().getPlugin("PlaceholderAPI") != null ?
+                PlaceholderAPI.setPlaceholders(Bukkit.getOfflinePlayer(message.getSender().getUniqueId()), parsedFormat)
```

Comment:
Does the chat work if PlaceholderAPI is not installed?

Reply:
Sure.